basic/kafka/src/main/java/org/springframework/integration/samples/kafka/Application.java

```diff
@@ -51,44 +55,55 @@
 import org.springframework.messaging.PollableChannel;
 import org.springframework.messaging.support.GenericMessage;
 
-import kafka.admin.AdminUtils;
-import kafka.utils.ZKStringSerializer$;
-import kafka.utils.ZkUtils;
-
 /**
  * @author Gary Russell
  * @since 4.2
  */
 @SpringBootApplication
+@EnableConfigurationProperties(KafkaAppProperties.class)
 public class Application {
 
-	@Value("${kafka.topic}")
-	private String topic;
-
-	@Value("${kafka.messageKey}")
-	private String messageKey;
-
-	@Value("${kafka.zookeeper.connect}")
-	private String zookeeperConnect;
+	@Autowired
+	private KafkaAppProperties properties;
 
 	public static void main(String[] args) throws Exception {
 		ConfigurableApplicationContext context
 				= new SpringApplicationBuilder(Application.class)
-				.web(false)
-				.run(args);
+					.web(WebApplicationType.NONE)
+					.run(args);
+		context.getBean(Application.class).runDemo(context);
+		context.close();
+		System.exit(0);
```

Comment:
I don't remember why it's there but I just tested without it and there are no problems.